Dockerfile

```diff
@@ -50,12 +63,12 @@ ENV PATH="/opt/venv/bin:$PATH"
 
 COPY --chown=sync-engine:sync-engine ./ ./
 RUN \
-  virtualenv /opt/venv && \
-  pip install setuptools==44.0.0 && \
-  pip install pip==20.3.4 && \
-  pip install --no-deps -r requirements_frozen.txt && \
-  pip install -e . && \
-  pip check
+  python"${PYTHON_VERSION}" -m virtualenv /opt/venv && \
+  /opt/venv/bin/python"${PYTHON_VERSION}" -m pip install setuptools==44.0.0 pip==20.3.4 && \
```

Comment:
I prefixed everything with exact Python versions to be clear that the right version of Python is invoked